### PR TITLE
[NATS] Add connections data stream dimensions and metric_type

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds dimensions and metric_type for the single metric in the connections data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #7745

## Screenshots
scrnli_9_21_2023_3-01-07 PM

### CODE REVIEW

**Comment by gpop63** [comment]
<details>
<summary>Rollover test</summary>

```json
{
  "acknowledged": true,
  "shards_acknowledged": true,
  "old_index": ".ds-metrics-nats.connections-default-2023.09.21-000001",
  "new_index": ".ds-metrics-nats.connections-default-2023.09.21-000002",
  "rolled_over": true,
  "dry_run": false,
  "conditions": {}
}
```

</details> 

<details>
<summary>Mappings</summary>

```json
  "nats": {
          "properties": {
            "connections": {
              "properties": {
                "total": {
                  "type": "long",
                  "time_series_metric": "gauge"
                }
              }
            },
```

</details>

File: packages/nats/data_stream/connections/fields/ecs.yml [hunk 1]
```diff
@@ -2,5 +2,15 @@
...
   name: service.type
+- external: ecs
+  name: agent.id
+  dimension: true
+- external: ecs
+  name: container.id
+  dimension: true
+- external: ecs
+  name: host.name
+  dimension: true
```
Comment (agithomas) [inline packages/nats/data_stream/connections/fields/ecs.yml:16]: NATS can be deployed nearly anywhere; on bare metal, in a VM, as a container, inside K8S, on a device, or whichever environment you choose.

Due to this, i would recommend having all common dimension fields included in the list. Please have a look at the TSDB migration guidelines document.

**Comment by agithomas** [comment]
@gpop63 , kindy fill the checklist in the PR description.

File: packages/nats/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.3.4"
+  changes:
+    - description: Add connections data stream dimensions and metric types.
```
Comment (agithomas) [inline packages/nats/changelog.yml:4]: Can you consider rephrasing the changelog as - Added dimensions and metric_type mapping to connections datastream? or similar?

**Review by agithomas — APPROVED** [review]
LGTM from a TSDB point of view.

**Review by constanca-m — APPROVED** [review]
Be careful to merge this after change `1.3.3` since the changelog is already on `1.3.4` here.

File: packages/nats/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.3.4"
+  changes:
+    - description: Added dimensions and metric_type mappings to connections data stream.
```
Comment (agithomas) [inline packages/nats/changelog.yml:4]:
```suggestion
    - description: Add dimensions and metric_type mappings to connections data stream.
```

**Pull request merged**